.github/workflows/spread-tests.yaml

```
@@ -101,32 +131,41 @@ jobs:
     - name: Setup run tests variable
       if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread')"
       run: |
+          get_new_tasks() {
+            local prefix=$1
+            local changes_param=$2
+            # The tests are just filtered when the change is a PR
+            # When 'Run Nested' label is added in a PR, all the nested tests have to be executed
+            TASKS_TO_RUN=""
+            if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ] || [ -z "${{ inputs.rules }}" ]; then
+                for TASKS in ${{ inputs.tasks }}; do
+                    TASKS_TO_RUN="$TASKS_TO_RUN $prefix:$TASKS"
+                done
+            else
+                NEW_TASKS="$(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ inputs.rules }}.yaml -p "$prefix" $changes_param)"
+                TASKS_TO_RUN="$NEW_TASKS"
+            fi
+            echo "$TASKS_TO_RUN"
+          }
+
+          CHANGES_PARAM=""
+          for CHANGE in $CHANGED_FILES; do
+              CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
+          done
           RUN_TESTS=""
-          SUGGESTED_TESTS=""
           # Save previous failed test results in FAILED_TESTS env var
           if [ -n "$FAILED_TESTS" ]; then
               RUN_TESTS="$FAILED_TESTS"
-          else
-              CHANGES_PARAM=""
-              for CHANGE in $CHANGED_FILES; do
-                  CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
-              done
+          elif [ "${{ inputs.systems }}" == "ALL" ]; then
```

Nitpick, = is good enough for comparison in shell.
```suggestion
          elif [ "${{ inputs.systems }}" = "ALL" ]; then
```